src/features/access-package/edit/edit_access_package_container.jsx

```diff
@@ -0,0 +1,189 @@
+        bottom: "auto",
+        right: 50,
+        position: "absolute"
+    }
+}));
+
+const EditAccessPackageContainer = (props) => {
+    const {
+        open, handleClose, handleSaveAccess, setEditOpen, openSave, handleSaveClose,
+        setSnackBarOpen, setSnackBarMessage
+    } = props;
+    const classes = useStyles();
+    const [componentSelectorOpen, setComponentSelectorOpen] = useState(false);
+    const [openCloseDialog, setOpenCloseDialog] = useState(false);
+    const selectedForEditingId = useSelector(state => state.access_package.selectedForEditing);
+    const accessPackages = useSelector(state => state.access_package.accessPackages);
+    const componentConfiguration = useSelector(state => state.component_configuration.componentConfiguration);
+    const [tabValue, setTabValue] = useState(0);
+    const dispatch = useDispatch();
+
+    let selectedAccessPackage = undefined;
+    accessPackages.map(ap => {
+        if (ap.dn === selectedForEditingId) {
+            selectedAccessPackage = ap;
+        }
+        return ap;
+    });
+
+    function openComponentSelector() {
+        setComponentSelectorOpen(true);
+    }
+
+    function handleCloseComponentSelector() {
+        setComponentSelectorOpen(false);
+    }
+
+    function findIndex(array, value) {
+        for (let i = 0; i < array.length; i += 1) {
+            if (array[i].dn === value.dn) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    function findComponentIndex(array, value) {
+        for (let i = 0; i < array.length; i += 1) {
+            if (array[i] === value) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    function removePathsFromList(list, component) {
+        return list.filter(path => {
+            let keepPath = true;
+            component.classes.map(aClass => {
+                if (aClass.path === path) {
+                    keepPath = false;
+                }
+                return null;
+            });
+            return keepPath;
+        });
+    }
+
+    function chooseComponent(component) {
+        let newAccessPackages = [...accessPackages];
+        let newAccessPackage = {...selectedAccessPackage};
+        const accessPackageIndex = findIndex(newAccessPackages, newAccessPackage);
+
+        if (newAccessPackage.components.includes(component.dn)) {
+            let componentIndex = findComponentIndex(newAccessPackage.components, component.dn);
+            newAccessPackage.components.splice(componentIndex, 1);
+            newAccessPackage.collection = removePathsFromList(newAccessPackage.collection, component);
+            newAccessPackage.read = removePathsFromList(newAccessPackage.read, component);
+            newAccessPackage.modify = removePathsFromList(newAccessPackage.modify, component);
+        } else {
+            newAccessPackage.components.push(component.dn);
+        }
+        newAccessPackages[accessPackageIndex] = newAccessPackage;
```

Comment:
may be -1

Reply:
Fixed by using array indexOf and filters. removed method